pkg/cache/cluster.go

```diff
@@ -53,6 +54,11 @@ const (
 	// Limit is required to avoid memory spikes during cache initialization.
 	// The default limit of 50 is chosen based on experiments.
 	defaultListSemaphoreWeight = 50
+
+	// Set ENABLE_CLUSTER_RESOURCE_CHILDREN to true to show namespace scoped
+	// resources owned by cluster scoped resources. By default, this feature is
+	// disabled due to performance implications.
+	enableClusterResourceChildrenEnv = "ENABLE_CLUSTER_RESOURCE_CHILDREN"
```

Comment:
Better not introducing an env variable here. I would recommend to pass this feature flag as part of the opts parameter in the `NewClusterCache` function:
```go
func NewClusterCache(config *rest.Config, opts ...UpdateSettingsFunc) *clusterCache
```

Reply:
I've removed the env variable and introduced an option for ClusterCache.